Index: icedtea-2.1.4/Makefile.am
===================================================================
--- icedtea-2.1.4.orig/Makefile.am	2013-01-26 18:20:55.152561203 +0100
+++ icedtea-2.1.4/Makefile.am	2013-01-26 18:41:07.637992206 +0100
@@ -1994,6 +1994,8 @@
 if BUILD_CACAO
 if !USE_SYSTEM_CACAO
 	cd cacao/cacao && \
+	patch -p1 -N < $(abs_top_srcdir)/patches/cacao-loadavg.patch && \
+        patch -p1 -N < $(abs_top_srcdir)/patches/cacao-arm-ucontext.patch && \
 	rm -f libtool && \
 	./autogen.sh && \
 	$(ARCH_PREFIX) ./configure \
