Index: icedtea-2.1.6/Makefile.am
===================================================================
--- icedtea-2.1.6.orig/Makefile.am	2013-02-23 21:19:13.000000000 +0100
+++ icedtea-2.1.6/Makefile.am	2013-02-23 21:29:29.891942491 +0100
@@ -18,7 +18,7 @@
 JAXWS_SHA256SUM = d8acabf54c1a4fe02e45ac4bdfe1ff9f3e7a3abc12884fcacd580bce9063a7c5
 JDK_SHA256SUM = 7eeee7d8479f97ab0eb66be3845224b8b4073de8829ed819175faf9ca2f3b5ca
 LANGTOOLS_SHA256SUM = 47db36264b345939176d1a5f36df84ad56d7fae76ce647ea1b619f7df0eb1eda
-OPENJDK_SHA256SUM = a856008db052b7f7ec4b466eee117fcc72229531136f71d4dd712c2f6e71cbd6
+OPENJDK_SHA256SUM = 9bdf6a874cb7ad4a5cbebb21cef9bebe8c0b8463a1c155df916eb311da9ca7a0
 
 CACAO_VERSION = a567bcb7f589
 CACAO_SHA256SUM = d49f79debc131a5694cae6ab3ba2864e7f3249ee8d9dc09aae8afdd4dc6b09f9
