Index: icedtea-2.1/acinclude.m4
===================================================================
--- icedtea-2.1.orig/acinclude.m4
+++ icedtea-2.1/acinclude.m4
@@ -909,6 +909,21 @@ AC_DEFUN([IT_WITH_PROJECT],
   AM_CONDITIONAL([USE_JDK7], test x"${project}" = "xjdk7")
 ])
 
+AC_DEFUN([AC_CHECK_WITH_CC_FOR_BUILD],
+[
+  AC_MSG_CHECKING(for compiler used for subsidiary programs)
+  AC_ARG_WITH([cc-for-build],
+              [AS_HELP_STRING(--with-cc-for-build,specify the compiler for subsidiary (helper) programs)],
+  [
+    CC_FOR_BUILD="${withval}"
+  ],
+  [
+    CC_FOR_BUILD="\$(CC)"
+  ])
+  AC_MSG_RESULT(${CC_FOR_BUILD})
+  AC_SUBST(CC_FOR_BUILD)
+])
+
 AC_DEFUN([IT_WITH_GCJ],
 [
   AC_MSG_CHECKING([whether to compile ecj natively])
Index: icedtea-2.1/configure.ac
===================================================================
--- icedtea-2.1.orig/configure.ac
+++ icedtea-2.1/configure.ac
@@ -18,6 +18,8 @@ AC_PREFIX_DEFAULT([bootstrap])
 AC_PROG_CC
 AC_PROG_CXX
 
+AC_CHECK_WITH_CC_FOR_BUILD
+
 IT_FIND_TOOL([MAKE], [make])
 IT_FIND_TOOL([GZIP], [gzip])
 IT_FIND_TOOL([ANT], [ant])
Index: icedtea-2.1/Makefile.am
===================================================================
--- icedtea-2.1.orig/Makefile.am
+++ icedtea-2.1/Makefile.am
@@ -416,6 +416,7 @@ ICEDTEA_ENV = \
 	JAVAC="" \
 	JAVA_HOME="" \
 	JDK_HOME="" \
+	CC_FOR_BUILD="$(CC_FOR_BUILD)" \
 	OE_CFLAGS="$(OE_CFLAGS)" \
 	OE_CPPFLAGS="$(OE_CPPFLAGS)" \
 	OE_CXXFLAGS="$(OE_CXXFLAGS)" \
