From 209095ebd9cbc57a04cbb66e4ff0448c8d120b19 Mon Sep 17 00:00:00 2001
From: Jaska Uimonen <jaska.uimonen@helsinki.fi>
Date: Wed, 12 Aug 2015 15:27:38 +0300
Subject: [PATCH 1/2] poptint.c: fix possible resource leak

Upstream-Status: Inappropriate [Fixed in upstream rewrite, can't backport]

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>
---
 poptint.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/poptint.c b/poptint.c
index 83fa7d6..e6454ec 100644
--- a/poptint.c
+++ b/poptint.c
@@ -103,8 +103,10 @@ strdup_locale_from_utf8 (/*@null@*/ char * istr)
 	size_t ob = db;
 	size_t err;
 
-	if (dstr == NULL)
-	    return NULL;
+	if (dstr == NULL) {
+		(void) iconv_close(cd);
+		return NULL;
+	}
 	err = iconv(cd, NULL, NULL, NULL, NULL);
 	while (1) {
 	    *pout = '\0';
-- 
2.1.0

