From 29dd41d83962f0d58a27a4ab10c5802556364758 Mon Sep 17 00:00:00 2001
From: Jaska Uimonen <jaska.uimonen@helsinki.fi>
Date: Wed, 12 Aug 2015 15:29:26 +0300
Subject: [PATCH 2/2] poptconfig.c: fix possible resource leak

Upstream-Status: Inappropriate [Fixed in upstream rewrite, can't backport]

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>
---
 poptconfig.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/poptconfig.c b/poptconfig.c
index 7840d0c..0b691e0 100644
--- a/poptconfig.c
+++ b/poptconfig.c
@@ -168,7 +168,7 @@ int poptReadFile(const char * fn, char ** bp, size_t * nbp, int flags)
     if (fdno < 0)
 	goto exit;
 
-    if ((nb = lseek(fdno, 0, SEEK_END)) == (off_t)-1
+    if ((nb = lseek(fdno, 0, SEEK_END)) <= (off_t)0
      || lseek(fdno, 0, SEEK_SET) == (off_t)-1
      || (b = calloc(sizeof(*b), (size_t)nb + 1)) == NULL
      || read(fdno, (char *)b, (size_t)nb) != (ssize_t)nb)
-- 
2.1.0

