--- jdk/make/GenerateSources.gmk.orig	2015-08-24 11:23:34.720648954 +0200
+++ jdk/make/GenerateSources.gmk	2015-08-24 11:24:41.376649205 +0200
@@ -73,9 +73,11 @@
     GENSRC += $(GENSRC_OSX_ICONS)
   endif

-  ifneq ($(OPENJDK_TARGET_OS), macosx)
-    include gensrc/GensrcX11Wrappers.gmk
-    GENSRC += $(GENSRC_X11WRAPPERS)
+  ifndef BUILD_HEADLESS_ONLY
+    ifneq ($(OPENJDK_TARGET_OS), macosx)
+      include gensrc/GensrcX11Wrappers.gmk
+      GENSRC += $(GENSRC_X11WRAPPERS)
+    endif
   endif
 endif

