diff --git openjdk/hotspot/make/linux/makefiles/gcc.make openjdk/hotspot/make/linux/makefiles/gcc.make
index 42f6843..765fc6d 100644
--- openjdk/hotspot/make/linux/makefiles/gcc.make
+++ openjdk/hotspot/make/linux/makefiles/gcc.make
@@ -149,11 +149,6 @@ else
   CFLAGS += -pipe
 endif
 
-# Compiler warnings are treated as errors
-ifneq ($(COMPILER_WARNINGS_FATAL),false)
-WARNINGS_ARE_ERRORS = -Werror
-endif
-
 # Except for a few acceptable ones
 # Since GCC 4.3, -Wconversion has changed its meanings to warn these implicit
 # conversions which might affect the values. To avoid that, we need to turn
