diff --git openjdk/jdk/make/com/sun/nio/sctp/Makefile openjdk/jdk/make/com/sun/nio/sctp/Makefile
index 4ae967f..fe669a9 100644
--- openjdk/jdk/make/com/sun/nio/sctp/Makefile
+++ openjdk/jdk/make/com/sun/nio/sctp/Makefile
@@ -64,11 +64,11 @@ ifneq ($(COMPILER_WARNINGS_FATAL),false)
 COMPILER_WARNINGS_FATAL=true
 endif
 #OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -ljava -lnet -lpthread -ldl
-OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -lnio -lnet -lpthread -ldl
+OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -lnio -ljavanet -lpthread -ldl
 endif
 ifeq ($(PLATFORM), solaris)
 #LIBSCTP = -lsctp
-OTHER_LDLIBS += $(LIBSOCKET) -L$(LIBDIR)/$(LIBARCH) -lnet -lnio
+OTHER_LDLIBS += $(LIBSOCKET) -L$(LIBDIR)/$(LIBARCH) -ljavanet -lnio
 endif # PLATFORM
 
 else # windows
diff --git openjdk/jdk/make/java/net/Makefile openjdk/jdk/make/java/net/Makefile
index 7283e45..720f3d3 100644
--- openjdk/jdk/make/java/net/Makefile
+++ openjdk/jdk/make/java/net/Makefile
@@ -25,7 +25,7 @@
 
 BUILDDIR = ../..
 PACKAGE = java.net
-LIBRARY = net
+LIBRARY = javanet
 PRODUCT = sun
 include $(BUILDDIR)/common/Defs.gmk
 
diff --git openjdk/jdk/make/java/nio/Makefile openjdk/jdk/make/java/nio/Makefile
index 191f17d..1e5f8e5 100644
--- openjdk/jdk/make/java/nio/Makefile
+++ openjdk/jdk/make/java/nio/Makefile
@@ -403,7 +403,7 @@
 endif
 
 ifeq ($(PLATFORM), linux)
-  OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -ljava -lnet -lpthread $(LIBDL)
+  OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -ljava -ljavanet -lpthread $(LIBDL)
   ifeq ($(SYSTEM_GIO), true)
     OTHER_LDLIBS += $(GIO_LIBS)
     OTHER_INCLUDES += $(GIO_CFLAGS) -DUSE_SYSTEM_GIO
@@ -413,15 +413,15 @@
 endif
 
 ifeq ($(PLATFORM), macosx)
-OTHER_LDLIBS += -L$(LIBDIR) -ljava -lnet -pthread -framework CoreFoundation
+OTHER_LDLIBS += -L$(LIBDIR) -ljava -ljavanet -pthread -framework CoreFoundation
 endif
 
 ifeq ($(PLATFORM), solaris)
 OTHER_LDLIBS += $(JVMLIB) $(LIBSOCKET) -lposix4 $(LIBDL) -lsendfile \
-		-L$(LIBDIR)/$(LIBARCH) -ljava -lnet
+		-L$(LIBDIR)/$(LIBARCH) -ljava -ljavanet
 endif
 ifeq ($(PLATFORM), aix)
-OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -ljava -lnet
+OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -ljava -ljavanet
 endif # PLATFORM
 
 ifeq ($(COMPILE_AGAINST_SYSCALLS), true)
diff --git openjdk/jdk/src/share/classes/java/net/AbstractPlainDatagramSocketImpl.java openjdk/jdk/src/share/classes/java/net/AbstractPlainDatagramSocketImpl.java
index b7f0f2f..b093868 100644
--- openjdk/jdk/src/share/classes/java/net/AbstractPlainDatagramSocketImpl.java
+++ openjdk/jdk/src/share/classes/java/net/AbstractPlainDatagramSocketImpl.java
@@ -69,7 +69,7 @@
      */
     static {
         java.security.AccessController.doPrivileged(
-                  new sun.security.action.LoadLibraryAction("net"));
+                  new sun.security.action.LoadLibraryAction("javanet"));
         init();
     }
 
diff --git openjdk/jdk/src/share/classes/java/net/AbstractPlainSocketImpl.java openjdk/jdk/src/share/classes/java/net/AbstractPlainSocketImpl.java
index c6a38ad..97de65b 100644
--- openjdk/jdk/src/share/classes/java/net/AbstractPlainSocketImpl.java
+++ openjdk/jdk/src/share/classes/java/net/AbstractPlainSocketImpl.java
@@ -78,7 +78,7 @@ abstract class AbstractPlainSocketImpl extends SocketImpl
      */
     static {
         java.security.AccessController.doPrivileged(
-                  new sun.security.action.LoadLibraryAction("net"));
+                  new sun.security.action.LoadLibraryAction("javanet"));
     }
 
     /**
diff --git openjdk/jdk/src/share/classes/java/net/DatagramPacket.java openjdk/jdk/src/share/classes/java/net/DatagramPacket.java
index 195ee86..6822a57 100644
--- openjdk/jdk/src/share/classes/java/net/DatagramPacket.java
+++ openjdk/jdk/src/share/classes/java/net/DatagramPacket.java
@@ -47,7 +47,7 @@ class DatagramPacket {
      */
     static {
         java.security.AccessController.doPrivileged(
-                  new sun.security.action.LoadLibraryAction("net"));
+                  new sun.security.action.LoadLibraryAction("javanet"));
         init();
     }
 
diff --git openjdk/jdk/src/share/classes/java/net/InetAddress.java openjdk/jdk/src/share/classes/java/net/InetAddress.java
index 3ea28e4..7c78eae 100644
--- openjdk/jdk/src/share/classes/java/net/InetAddress.java
+++ openjdk/jdk/src/share/classes/java/net/InetAddress.java
@@ -260,7 +260,7 @@ class InetAddress implements java.io.Serializable {
     static {
         preferIPv6Address = java.security.AccessController.doPrivileged(
             new GetBooleanAction("java.net.preferIPv6Addresses")).booleanValue();
-        AccessController.doPrivileged(new LoadLibraryAction("net"));
+        AccessController.doPrivileged(new LoadLibraryAction("javanet"));
         init();
     }
 
diff --git openjdk/jdk/src/share/classes/java/net/NetworkInterface.java openjdk/jdk/src/share/classes/java/net/NetworkInterface.java
index 6314d15..761fdda 100644
--- openjdk/jdk/src/share/classes/java/net/NetworkInterface.java
+++ openjdk/jdk/src/share/classes/java/net/NetworkInterface.java
@@ -53,7 +53,7 @@ public final class NetworkInterface {
     private static final int defaultIndex; /* index of defaultInterface */
 
     static {
-        AccessController.doPrivileged(new LoadLibraryAction("net"));
+        AccessController.doPrivileged(new LoadLibraryAction("javanet"));
         init();
         defaultInterface = DefaultInterface.getDefault();
         if (defaultInterface != null) {
diff --git openjdk/jdk/src/share/classes/sun/net/sdp/SdpSupport.java openjdk/jdk/src/share/classes/sun/net/sdp/SdpSupport.java
index 5baca6e..befd0b9 100644
--- openjdk/jdk/src/share/classes/sun/net/sdp/SdpSupport.java
+++ openjdk/jdk/src/share/classes/sun/net/sdp/SdpSupport.java
@@ -76,6 +76,6 @@ public final class SdpSupport {
 
     static {
         AccessController.doPrivileged(
-            new sun.security.action.LoadLibraryAction("net"));
+            new sun.security.action.LoadLibraryAction("javanet"));
     }
 }
diff --git openjdk/jdk/src/share/classes/sun/net/spi/DefaultProxySelector.java openjdk/jdk/src/share/classes/sun/net/spi/DefaultProxySelector.java
index d7aa358..9b2c719 100644
--- openjdk/jdk/src/share/classes/sun/net/spi/DefaultProxySelector.java
+++ openjdk/jdk/src/share/classes/sun/net/spi/DefaultProxySelector.java
@@ -95,7 +95,7 @@ public class DefaultProxySelector extends ProxySelector {
                 }});
         if (b != null && b.booleanValue()) {
             java.security.AccessController.doPrivileged(
-                      new sun.security.action.LoadLibraryAction("net"));
+                      new sun.security.action.LoadLibraryAction("javanet"));
             hasSystemProxies = init();
         }
     }
diff --git openjdk/jdk/src/share/classes/sun/nio/ch/Util.java openjdk/jdk/src/share/classes/sun/nio/ch/Util.java
index 10662df..34d441a 100644
--- openjdk/jdk/src/share/classes/sun/nio/ch/Util.java
+++ openjdk/jdk/src/share/classes/sun/nio/ch/Util.java
@@ -483,7 +483,7 @@ class Util {
                 return;
             loaded = true;
             java.security.AccessController
-                .doPrivileged(new sun.security.action.LoadLibraryAction("net"));
+                .doPrivileged(new sun.security.action.LoadLibraryAction("javanet"));
             java.security.AccessController
                 .doPrivileged(new sun.security.action.LoadLibraryAction("nio"));
             // IOUtil must be initialized; Its native methods are called from
diff --git openjdk/jdk/src/solaris/classes/sun/net/dns/ResolverConfigurationImpl.java openjdk/jdk/src/solaris/classes/sun/net/dns/ResolverConfigurationImpl.java
index 3695b5d..7c833cc 100644
--- openjdk/jdk/src/solaris/classes/sun/net/dns/ResolverConfigurationImpl.java
+++ openjdk/jdk/src/solaris/classes/sun/net/dns/ResolverConfigurationImpl.java
@@ -247,7 +247,7 @@ public class ResolverConfigurationImpl
 
     static {
         java.security.AccessController.doPrivileged(
-            new sun.security.action.LoadLibraryAction("net"));
+            new sun.security.action.LoadLibraryAction("javanet"));
     }
 
 }
diff --git openjdk/jdk/src/windows/classes/sun/net/dns/ResolverConfigurationImpl.java openjdk/jdk/src/windows/classes/sun/net/dns/ResolverConfigurationImpl.java
index 7970885..46ab112 100644
--- openjdk/jdk/src/windows/classes/sun/net/dns/ResolverConfigurationImpl.java
+++ openjdk/jdk/src/windows/classes/sun/net/dns/ResolverConfigurationImpl.java
@@ -159,7 +159,7 @@ public class ResolverConfigurationImpl
 
     static {
         java.security.AccessController.doPrivileged(
-            new sun.security.action.LoadLibraryAction("net"));
+            new sun.security.action.LoadLibraryAction("javanet"));
         init0();
 
         // start the address listener thread
--- openjdk/jdk/src/windows/classes/sun/net/PortConfig.java	2015-01-08 21:19:49.000000000 +0100
+++ openjdk/jdk/src/windows/classes/sun/net/PortConfig.java	2015-04-02 13:45:19.330199119 +0200
@@ -43,7 +43,7 @@
         AccessController.doPrivileged(
             new java.security.PrivilegedAction<Void>() {
                 public Void run() {
-                    System.loadLibrary("net");
+                    System.loadLibrary("javanet");
                     return null;
                 }
             });
--- openjdk/jdk/src/windows/classes/sun/nio/fs/RegistryFileTypeDetector.java	2015-01-08 21:19:49.000000000 +0100
+++ openjdk/jdk/src/windows/classes/sun/nio/fs/RegistryFileTypeDetector.java	2015-04-02 13:47:48.245530982 +0200
@@ -74,7 +74,7 @@
             @Override
             public Void run() {
                 // nio.dll has dependency on net.dll
-                System.loadLibrary("net");
+                System.loadLibrary("javanet");
                 System.loadLibrary("nio");
                 return null;
         }});
--- openjdk/jdk/src/windows/classes/sun/nio/fs/WindowsNativeDispatcher.java	2015-01-08 21:19:49.000000000 +0100
+++ openjdk/jdk/src/windows/classes/sun/nio/fs/WindowsNativeDispatcher.java	2015-04-02 13:48:46.363509265 +0200
@@ -1124,7 +1124,7 @@
         AccessController.doPrivileged(new PrivilegedAction<Void>() {
             public Void run() {
                 // nio.dll has dependency on net.dll
-                System.loadLibrary("net");
+                System.loadLibrary("javanet");
                 System.loadLibrary("nio");
                 return null;
         }});
--- openjdk/jdk/src/solaris/classes/sun/net/PortConfig.java	2015-01-08 21:19:49.000000000 +0100
+++ openjdk/jdk/src/solaris/classes/sun/net/PortConfig.java	2015-04-02 13:49:49.541999538 +0200
@@ -44,7 +44,7 @@
         AccessController.doPrivileged(
             new java.security.PrivilegedAction<Void>() {
                 public Void run() {
-                    System.loadLibrary("net");
+                    System.loadLibrary("javanet");
                     String os = System.getProperty("os.name");
                     if (os.startsWith("Linux")) {
                         defaultLower = 32768;
