From: Mario Domenech Goulart <mario@ossystems.com.br>
Organization: O.S. Systems Software LTDA.

Upstream-Status: Pending
---
--- openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2015-03-30 13:43:43.433168285 -0300
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2015-03-30 13:45:57.601833592 -0300
@@ -410,9 +410,9 @@
  *        7: The default directories, normally /lib and /usr/lib.
  */
 #if defined(AMD64) || defined(_LP64) && (defined(SPARC) || defined(PPC) || defined(S390) || defined(AARCH64)) || defined(BUILTIN_SIM)
-#define DEFAULT_LIBPATH "/usr/lib64:/lib64:/lib:/usr/lib"
+#define DEFAULT_LIBPATH "/usr/lib64:/lib64:/lib:/usr/lib:/usr/lib64/jni:/usr/lib/jni"
 #else
-#define DEFAULT_LIBPATH "/lib:/usr/lib"
+#define DEFAULT_LIBPATH "/lib:/usr/lib:/usr/lib/jni"
 #endif
 
 #define EXTENSIONS_DIR  "/lib/ext"
