Index: icedtea-2.1/Makefile.am
===================================================================
--- icedtea-2.1.orig/Makefile.am
+++ icedtea-2.1/Makefile.am
@@ -413,6 +413,10 @@ ICEDTEA_ENV = \
 	FT2_LIBS="$(FREETYPE2_LIBS)" \
 	ALT_PARALLEL_COMPILE_JOBS="$(PARALLEL_JOBS)" \
 	HOTSPOT_BUILD_JOBS="$(PARALLEL_JOBS)" \
+        OE_CFLAGS="$(OE_CFLAGS)" \
+        OE_CPPFLAGS="$(OE_CPPFLAGS)" \
+        OE_CXXFLAGS="$(OE_CXXFLAGS)" \
+        OE_LDFLAGS="$(OE_LDFLAGS)" \
 	JAVAC="" \
 	JAVA_HOME="" \
 	JDK_HOME="" \
@@ -428,10 +432,7 @@ ICEDTEA_ENV = \
 	VERBOSE="$(VERBOSE)" \
 	STATIC_CXX="false" \
 	BUILD_GCC="$(CC)" \
-	BUILD_CPP="$(CXX)" \
-	USE_SYSTEM_CUPS="true" \
-	CUPS_LIBS="${CUPS_LIBS}" \
-	CUPS_CFLAGS="${CUPS_CFLAGS}"
+	BUILD_CPP="$(CXX)"
 
 if ENABLE_CACAO
 ICEDTEA_ENV += \
@@ -504,12 +505,8 @@ ICEDTEA_ENV += \
 	GIF_CFLAGS="${GIF_CFLAGS}"
 endif
 
-if USE_SYSTEM_GTK
 ICEDTEA_ENV += \
-	USE_SYSTEM_GTK="true" \
-	GTK_LIBS="${GTK_LIBS}" \
-	GTK_CFLAGS="${GTK_CFLAGS}"
-endif
+	USE_SYSTEM_GTK="false"
 
 if USE_SYSTEM_GIO
 ICEDTEA_ENV += \
