diff --git openjdk/jdk/make/common/Subdirs.gmk openjdk/jdk/make/common/Subdirs.gmk
index f212d50..da6e9e1 100644
--- openjdk/jdk/make/common/Subdirs.gmk
+++ openjdk/jdk/make/common/Subdirs.gmk
@@ -88,28 +88,6 @@ done
 @$(ECHO) "Done Processing OTHERSUBDIRS: $(OTHERSUBDIRS)"
 endef
 
-#
-# Iterate the list specified in SUBDIRS_<group> only if
-# SUBDIRS_<group> is set.
-#
-ifdef SUBDIRS_desktop 
-  define subdirs-desktop-loop
-      @$(call subdirs-group-loop,SUBDIRS_desktop)
-  endef
-else   
-  define subdirs-desktop-loop
-  endef
-endif   # SUBDIRS_desktop 
-
-ifdef SUBDIRS_enterprise 
-  define subdirs-enterprise-loop
-      @$(call subdirs-group-loop,SUBDIRS_enterprise)
-  endef
-else   
-define subdirs-enterprise-loop
-endef
-endif   # SUBDIRS_enterprise 
-
 ifdef SUBDIRS_management 
   define subdirs-management-loop
       @$(call subdirs-group-loop,SUBDIRS_management)
@@ -140,7 +118,7 @@ endif   # SUBDIRS_tools
 #
 # If BUILD_MODULES is not set or it's set to "all",
 # iterate all groups.
-SUBDIRS_all = $(SUBDIRS) $(SUBDIRS_desktop) $(SUBDIRS_enterprise) \
+SUBDIRS_all = $(SUBDIRS) $(SUBDIRS_enterprise) \
 	      $(SUBDIRS_management) $(SUBDIRS_misc) $(SUBDIRS_tools)
 
 define SUBDIRS-loop
diff --git a/jdk/src/solaris/native/sun/java2d/x11/XRSurfaceData.c b/jdk/src/solaris/native/sun/java2d/x11/XRSurfaceData.c
index 5de53ca..b598a22 100644
--- a/jdk/src/solaris/native/sun/java2d/x11/XRSurfaceData.c
+++ b/jdk/src/solaris/native/sun/java2d/x11/XRSurfaceData.c
@@ -29,7 +29,7 @@
 #include "X11SurfaceData.h"
 
 /*#include <xcb/xcb.h>*/
-#include <Xrender.h>
+#include <X11/extensions/Xrender.h>
 
 #ifndef RepeatNone  /* added in 0.10 */
 #define RepeatNone    0
diff --git a/jdk/make/sun/xawt/FILES_c_unix.gmk b/jdk/make/sun/xawt/FILES_c_unix.gmk
index 57b34c3..c629f78 100644
--- a/jdk/make/sun/xawt/FILES_c_unix.gmk
+++ b/jdk/make/sun/xawt/FILES_c_unix.gmk
@@ -84,15 +84,8 @@ FILES_c = \
 	XRSurfaceData.c \
 	XRBackendNative.c
 
-ifndef USE_SYSTEM_GTK
 FILES_c += \
 	gtk_fp.c
-endif
-
-ifndef USE_SYSTEM_CUPS
-FILES_c += \
-	cups_fp.c
-endif
 
 ifndef USE_SYSTEM_FONTCONFIG
 FILES_c += \
diff --git a/jdk/make/sun/awt/FILES_c_unix.gmk b/jdk/make/sun/awt/FILES_c_unix.gmk
index dc1f61e..489abc1 100644
--- a/jdk/make/sun/awt/FILES_c_unix.gmk
+++ b/jdk/make/sun/awt/FILES_c_unix.gmk
@@ -230,11 +230,6 @@ FILES_NO_MOTIF_c = \
         AccelGlyphCache.c \
 	CUPSfuncs.c
 
-ifndef USE_SYSTEM_CUPS
-FILES_NO_MOTIF_c += \
-	cups_fp.c
-endif
-
 ifndef USE_SYSTEM_FONTCONFIG
 FILES_NO_MOTIF_c += \
 	fontconfig_fp.c
