diff --git a/configure.in b/configure.in
index 448a292..31ca001 100644
--- a/configure.in
+++ b/configure.in
@@ -1382,7 +1382,7 @@ if test $cross_compiling = no; then
    BUILD_CFLAGS="$CFLAGS $CPPFLAGS $INCLUDES -DHAVE_CONFIG_H"
    BUILD_LDFLAGS="$LDFLAGS"
 else
-   BUILD_CFLAGS=
+   BUILD_CFLAGS='-I. -I$(top_builddir)/lib -I$(top_srcdir)/lib -DHAVE_CONFIG_H'
    BUILD_LDFLAGS=
 fi
 AC_SUBST(BUILD_CFLAGS)
