Index: grub-1.98/conf/any-emu.mk
===================================================================
--- grub-1.98.orig/conf/any-emu.mk	2011-02-24 00:54:20.825116207 -0600
+++ grub-1.98/conf/any-emu.mk	2011-02-24 00:55:55.465116194 -0600
@@ -820,7 +820,7 @@
 	./gentrigtables > $@
 DISTCLEANFILES += trigtables.c
 gentrigtables: gentrigtables.c
-	$(CC) -o $@ $^ $(CPPFLAGS) -lm
+	$(BUILD_CC) -o $@ $^ $(BUILD_CPPFLAGS) -I${top_srcdir}/include -lm
 DISTCLEANFILES += gentrigtables
 
 # For grub-mkfont.
Index: grub-1.98/conf/any-emu.rmk
===================================================================
--- grub-1.98.orig/conf/any-emu.rmk	2011-02-24 00:54:24.565116208 -0600
+++ grub-1.98/conf/any-emu.rmk	2011-02-24 00:56:40.665116186 -0600
@@ -110,7 +110,7 @@
 	./gentrigtables > $@
 DISTCLEANFILES += trigtables.c
 gentrigtables: gentrigtables.c
-	$(CC) -o $@ $^ $(CPPFLAGS) -lm
+	$(BUILD_CC) -o $@ $^ $(BUILD_CPPFLAGS) -I${top_srcdir}/include -lm
 DISTCLEANFILES += gentrigtables
 
 # For grub-mkfont.
Index: grub-1.98/conf/common.mk
===================================================================
--- grub-1.98.orig/conf/common.mk	2011-02-24 00:54:30.345116207 -0600
+++ grub-1.98/conf/common.mk	2011-02-24 00:57:08.415116181 -0600
@@ -11159,7 +11159,7 @@
 	./gentrigtables > $@
 DISTCLEANFILES += trigtables.c
 gentrigtables: gentrigtables.c
-	$(CC) -o $@ $^ $(CPPFLAGS) -lm
+	$(BUILD_CC) -o $@ $^ $(BUILD_CPPFLAGS) -I${top_srcdir}/include -lm
 DISTCLEANFILES += gentrigtables
 
 pkglib_MODULES += setjmp.mod
Index: grub-1.98/conf/common.rmk
===================================================================
--- grub-1.98.orig/conf/common.rmk	2011-02-24 00:54:34.035116205 -0600
+++ grub-1.98/conf/common.rmk	2011-02-24 00:57:31.045116179 -0600
@@ -736,7 +736,7 @@
 	./gentrigtables > $@
 DISTCLEANFILES += trigtables.c
 gentrigtables: gentrigtables.c
-	$(CC) -o $@ $^ $(CPPFLAGS) -lm
+	$(BUILD_CC) -o $@ $^ $(BUILD_CPPFLAGS) -I${top_srcdir}/include -lm
 DISTCLEANFILES += gentrigtables
 
 pkglib_MODULES += setjmp.mod
