Index: grub-1.98/util/grub-install.in
===================================================================
--- grub-1.98.orig/util/grub-install.in	2011-01-25 23:47:05.587521825 -0600
+++ grub-1.98/util/grub-install.in	2011-01-25 23:47:23.867521822 -0600
@@ -306,7 +306,7 @@
 
 if [ "x${devabstraction_module}" = "x" ] ; then
     if [ x"${install_device}" != x ]; then
-      if echo "${install_device}" | grep -qx "(.*)" ; then
+      if echo "${install_device}" | grep -q "(.*)" ; then
         install_drive="${install_device}"
       else
         install_drive="`$grub_probe --target=drive --device ${install_device}`"
