--- hotspot/src/os/linux/vm/os_linux.cpp	2015-03-07 10:25:19.000000000 -0800
+++ hotspot/src/os/linux/vm/os_linux.cpp	2015-03-07 10:25:19.000000000 -0800
+++ a.cpp	2015-07-23 11:45:17.666243540 -0700
@@ -4847,7 +4847,7 @@
   Linux::capture_initial_stack(JavaThread::stack_size_at_create());
 
 #if defined(IA32)
-  workaround_expand_exec_shield_cs_limit();
+//  workaround_expand_exec_shield_cs_limit();
 #endif
 
   Linux::libpthread_init();
