From 9257b72a6183b0e58b3caef388aa46c1c9a945fc Mon Sep 17 00:00:00 2001
From: Brendan Le Foll <brendan.le.foll@intel.com>
Date: Thu, 14 Aug 2014 18:06:29 +0100
Subject: [PATCH] SeeedOLED.h: fix Wire.h include

Signed-off-by: Brendan Le Foll <brendan.le.foll@intel.com>
---
 src/libraries/SeeedOLED.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libraries/SeeedOLED.h b/src/libraries/SeeedOLED.h
index 49f3012..7298fd7 100644
--- a/src/libraries/SeeedOLED.h
+++ b/src/libraries/SeeedOLED.h
@@ -24,7 +24,7 @@
 
 // SeeedOLED Instruction set addresses
 
-#include <Wyliodrin/Wire.h>
+#include "Wire.h"
 
 
 #define pgm_read_byte(addr) (*(const unsigned char *)(addr))
-- 
2.0.1

