diff -Nurd openjdk/hotspot/src/share/vm/opto/loopnode.cpp openjdk/hotspot/src/share/vm/opto/loopnode.cpp
--- openjdk/hotspot/src/share/vm/opto/loopnode.cpp	2012-10-05 14:18:19.725747261 +0200
+++ openjdk/hotspot/src/share/vm/opto/loopnode.cpp	2012-10-05 14:18:40.751035242 +0200
@@ -534,7 +534,7 @@
 Node* CountedLoopNode::match_incr_with_optional_truncation(
                       Node* expr, Node** trunc1, Node** trunc2, const TypeInt** trunc_type) {
   // Quick cutouts:
-  if (expr == NULL || expr->req() != 3)  return false;
+  if (expr == NULL || expr->req() != 3)  return NULL;
 
   Node *t1 = NULL;
   Node *t2 = NULL;
