Index: icedtea6-1.8.11/Makefile.am
===================================================================
--- icedtea6-1.8.11.orig/Makefile.am	2011-12-20 00:49:21.000000000 +0100
+++ icedtea6-1.8.11/Makefile.am	2011-12-20 01:07:08.431134312 +0100
@@ -1877,7 +1877,7 @@
 stamps/rewrite-rhino.stamp: stamps/rewriter.stamp $(RHINO_JAR)
 if WITH_RHINO
 	mkdir -p rhino/rhino.{old,new}
-	(cd rhino/rhino.old ; jar xf $(RHINO_JAR))
+	(cd rhino/rhino.old ; $(ICEDTEA_BOOT_DIR)/bin/jar xf $(RHINO_JAR))
 	$(ICEDTEA_BOOT_DIR)/bin/java -cp $(abs_top_builddir)/rewriter \
 	  com.redhat.rewriter.ClassRewriter \
 	  $(abs_top_builddir)/rhino/rhino.old $(abs_top_builddir)/rhino/rhino.new \
