diff --git a/build_common/SConscript b/build_common/SConscript
index 6bbdbde..256dcbd 100644
--- a/build_common/SConscript
+++ b/build_common/SConscript
@@ -324,6 +324,8 @@ if target_os == "yocto":
                         env[tool] = os.path.join(path, os.environ[tool])
                         break
         env['CROSS_COMPILE'] = target_prefix[:len(target_prefix) - 1]
+        env['ENV']['PKG_CONFIG_SYSROOT_DIR'] = os.environ['PKG_CONFIG_SYSROOT_DIR']
+        env['ENV']['PKG_CONFIG_PATH'] = os.environ['PKG_CONFIG_PATH']
     except:
         print "ERROR in Yocto cross-toolchain environment"
         Exit(1)
