From aba4418696ef31ab5afc9811903a044366619dc4 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Mon, 22 Feb 2016 22:07:08 -0800
Subject: [PATCH 2/4] Fix regression that introduced a directory named '/ '

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>

Upstream-Status: Backport (v3.0+)

---
 src/chroot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/chroot.c b/src/chroot.c
index f961d58..2878496 100644
--- a/src/chroot.c
+++ b/src/chroot.c
@@ -40,7 +40,7 @@ void chroot_create_full(int newversion)
 	char * command;
 	char *full_dir;
 
-	string_or_die(&full_dir, "%s/%i/full/ ", image_dir, newversion);
+	string_or_die(&full_dir, "%s/%i/full/", image_dir, newversion);
 
 	g_mkdir_with_parents(full_dir, S_IRWXU);
 
-- 
2.5.0

