Index: icedtea6-1.13.3/Makefile.am
===================================================================
--- icedtea6-1.13.3.orig/Makefile.am	2014-06-09 21:46:36.640603938 +0200
+++ icedtea6-1.13.3/Makefile.am	2014-06-09 21:51:34.228435112 +0200
@@ -795,6 +795,11 @@
 	JAVAC="" \
 	JAVA_HOME="" \
 	JDK_HOME="" \
+	OE_CFLAGS="$(OE_CFLAGS)" \
+	OE_CPPFLAGS="$(OE_CPPFLAGS)" \
+	OE_CXXFLAGS="$(OE_CXXFLAGS)" \
+	OE_LDFLAGS="$(OE_LDFLAGS)" \
+	OE_LAUNCHER_LDFLAGS="$(OE_LAUNCHER_LDFLAGS)" \
 	ANT_RESPECT_JAVA_HOME="TRUE" \
 	DISTRIBUTION_ID="$(DIST_ID)" \
 	DERIVATIVE_ID="$(ICEDTEA_NAME) $(PACKAGE_VERSION)$(ICEDTEA_REV)" \
@@ -2025,9 +2030,7 @@
 	rm -f stamps/pax-mark-vm-debug.stamp
 
 stamps/check-crypto.stamp: stamps/cryptocheck.stamp stamps/pax-mark-vm.stamp
-	if [ -e $(BUILD_OUTPUT_DIR)/j2sdk-image/bin/java ] ; then \
-	  $(BUILD_OUTPUT_DIR)/j2sdk-image/bin/java -cp $(CRYPTO_CHECK_BUILD_DIR) TestCryptoLevel ; \
-	fi
+	$(BOOT_DIR)/bin/java -cp $(CRYPTO_CHECK_BUILD_DIR) TestCryptoLevel ; \
 	mkdir -p stamps
 	touch $@
 
@@ -2035,9 +2038,7 @@
 	rm -f stamps/check-crypto.stamp
 
 stamps/check-crypto-debug.stamp: stamps/cryptocheck.stamp stamps/pax-mark-vm-debug.stamp
-	if [ -e $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/bin/java ] ; then \
-	  $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/bin/java -cp $(CRYPTO_CHECK_BUILD_DIR) TestCryptoLevel ; \
-	fi
+	$(BOOT_DIR)/bin/java -cp $(CRYPTO_CHECK_BUILD_DIR) TestCryptoLevel ; \
 	mkdir -p stamps
 	touch $@
 
@@ -2219,9 +2220,7 @@
 	rm -f stamps/pax-mark-vm-ecj.stamp
 
 stamps/check-crypto-boot.stamp: stamps/cryptocheck.stamp stamps/pax-mark-vm-ecj.stamp
-	if [ -e $(ECJ_BUILD_OUTPUT_DIR)/j2sdk-image/bin/java ] ; then \
-	  $(ECJ_BUILD_OUTPUT_DIR)/j2sdk-image/bin/java -cp $(CRYPTO_CHECK_BUILD_DIR) TestCryptoLevel ; \
-	fi
+	$(BOOT_DIR)/bin/java -cp $(CRYPTO_CHECK_BUILD_DIR) TestCryptoLevel ; \
 	mkdir -p stamps
 	touch $@
 
@@ -2379,10 +2378,14 @@
 if BUILD_JAMVM
 	cd jamvm/jamvm && \
 	LDFLAGS="-Xlinker -z -Xlinker noexecstack" \
-	./autogen.sh --with-java-runtime-library=openjdk6 \
-	  --prefix=$(abs_top_builddir)/jamvm/install ; \
-	$(MAKE) ; \
-	$(MAKE) install
+	$(ARCH_PREFIX) ./autogen.sh --with-java-runtime-library=openjdk6 \
+		--prefix=$(abs_top_builddir)/jamvm/install \
+		--host=$(host_alias) \
+		--build=$(build_alias) \
+		--target=$(target_alias)  \
+		--with-libtool-sysroot=${ALT_FREETYPE_LIB_PATH} ; \
+	$(ARCH_PREFIX) $(MAKE) ; \
+	$(ARCH_PREFIX) $(MAKE) install
 	mkdir -p $(abs_top_builddir)/jamvm/install/hotspot/jre/lib/$(INSTALL_ARCH_DIR)/server
 	cp $(abs_top_builddir)/jamvm/install/lib/libjvm.so $(abs_top_builddir)/jamvm/install/hotspot/jre/lib/$(INSTALL_ARCH_DIR)/server
 	ln -sf server $(abs_top_builddir)/jamvm/install/hotspot/jre/lib/$(INSTALL_ARCH_DIR)/client
@@ -2446,6 +2449,7 @@
 	  --host=$(host_alias) \
 	  --build=$(build_alias) \
 	  --target=$(target_alias) \
+	  --with-libtool-sysroot=${ALT_FREETYPE_LIB_PATH} \
 	  --prefix=$(abs_top_builddir)/cacao/install \
 	  --with-java-runtime-library=openjdk \
 	  --with-java-runtime-library-prefix=$(abs_top_builddir)/openjdk \
@@ -2514,7 +2518,7 @@
 # configure script arguments, quoted in single quotes
 CONFIGURE_ARGS = @CONFIGURE_ARGS@
 ADD_ZERO_CONFIGURE_ARGS = \
-	--with-jdk-home=$(BUILD_OUTPUT_DIR)/j2sdk-image \
+	--with-jdk-home=$(abs_top_builddir)/bootstrap/jdk1.6.0 \
 	--enable-zero
 if ADD_SHARK_BUILD
 ADD_ZERO_CONFIGURE_ARGS += \
Index: icedtea6-1.13.3/javac.in
===================================================================
--- icedtea6-1.13.3.orig/javac.in	2014-06-09 21:46:36.640603938 +0200
+++ icedtea6-1.13.3/javac.in	2014-06-09 21:46:36.636603946 +0200
@@ -40,7 +40,7 @@
 {
     my @new_args = @{$_[0]};
 
-    for my $opt (NO_DUP_ARGS) 
+    for my $opt (NO_DUP_ARGS)
     {
 	my @indices = reverse grep {$new_args[$_] eq $opt} 0..$#new_args;
 	if (@indices > 1) {
@@ -49,7 +49,7 @@
 	}
     }
 
-    for my $opt (STRIP_ARGS) 
+    for my $opt (STRIP_ARGS)
     {
 	my @indices = reverse grep {$new_args[$_] eq $opt} 0..$#new_args;
 	splice @new_args, $_, 1 for @indices;
@@ -91,7 +91,10 @@
     my @CLASSPATH = ('@ECJ_JAR@');
     push @CLASSPATH, split /:/, $ENV{"CLASSPATH"} if exists $ENV{"CLASSPATH"};
     $ENV{"CLASSPATH"} = join ':', @CLASSPATH;
-    exec '@JAVA@', @$vm_args, 'org.eclipse.jdt.internal.compiler.batch.Main', @ecj_parms, @$javac_args;
+
+    # Compiling hotspot-tools consumes tons of memory and exceeds any default
+    # limits for jamvm and cacao.
+    exec '@JAVA@', '-Xmx1024m', @$vm_args, 'org.eclipse.jdt.internal.compiler.batch.Main', @ecj_parms, @$javac_args;
 }
 else
 {
