diff --git openjdk/hotspot/make/linux/makefiles/rules.make openjdk/hotspot/make/linux/makefiles/rules.make
index b40c081..2dfbfc8 100644
--- openjdk/hotspot/make/linux/makefiles/rules.make
+++ openjdk/hotspot/make/linux/makefiles/rules.make
@@ -36,6 +36,7 @@ DEMANGLE        = $(DEMANGLER) < $@ > .$@ && mv -f .$@ $@
 # variables: $(CFLAGS), $(CXXFLAGS) and $(CPPFLAGS).
 CC_COMPILE       = $(CC) $(CXXFLAGS)
 CXX_COMPILE      = $(CXX) $(CXXFLAGS) $(CFLAGS)
+CC_HOST_COMPILE  = $(BUILD_CC) $(CPPFLAGS) $(CFLAGS)
 
 AS.S            = $(AS) $(ASFLAGS)
 
