From 418709f1b216bbef6e2d815a784e15bc9eca17ac Mon Sep 17 00:00:00 2001
From: Brendan Le Foll <brendan.le.foll@intel.com>
Date: Tue, 16 Jun 2015 13:42:28 +0100
Subject: [PATCH] Makefile.am: Rempve setting of CFLAGS user var

Signed-off-by: Brendan Le Foll <brendan.le.foll@intel.com>
---
 Makefile.am |    2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 788d37c..d64c081 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,5 @@
 AUTOMAKE_OPTIONS = subdir-objects
 
-CFLAGS = -g -Wall
-
 PARSER_CFLAGS=@PARSER_CFLAGS@
 PARSER_LIBS=@PARSER_LIBS@
 
-- 
1.7.10.4

