diff --git a/src/jam.c b/src/jam.c
index b76ce8b..dc5eac3 100644
--- a/src/jam.c
+++ b/src/jam.c
@@ -133,9 +133,7 @@ int parseCommandLine(int argc, char *argv[], InitArgs *args) {
     for(i = 1; i < argc; i++) {
         if(*argv[i] != '-') {
             if(args->min_heap > args->max_heap) {
-                printf("Minimum heap size greater than max!\n");
-                status = 1;
-                goto exit;
+                args->min_heap = args->max_heap;
             }
 
             if(args->props_count) {
