Index: icedtea6-1.11.5/Makefile.am
===================================================================
--- icedtea6-1.11.5.orig/Makefile.am	2013-01-27 21:47:24.609119623 +0100
+++ icedtea6-1.11.5/Makefile.am	2013-01-27 21:49:31.250516775 +0100
@@ -1993,6 +1993,8 @@
 if BUILD_CACAO
 if !USE_SYSTEM_CACAO
 	cd cacao/cacao && \
+	patch -p1 -N < $(abs_top_srcdir)/patches/cacao-loadavg.patch && \
+	patch -p1 -N < $(abs_top_srcdir)/patches/cacao-arm-ucontext.patch && \
 	rm -f libtool && \
 	./autogen.sh && \
 	$(ARCH_PREFIX) ./configure \
