diff --git openjdk/jdk/make/com/sun/jmx/Makefile openjdk/jdk/make/com/sun/jmx/Makefile
index bdc1f8f..76f0c00 100644
--- openjdk/jdk/make/com/sun/jmx/Makefile
+++ openjdk/jdk/make/com/sun/jmx/Makefile
@@ -102,7 +102,7 @@ FILES_ties   = $(subst javax/management/remote/rmi/,javax/management/remote/rmi/
 # when available, we need to run with latest rmic version available. rmic 
 # launch tool not built at this stage but we can invoke via rmi class.
 
-RMIC_JAVA = $(OUTPUTDIR)/bin/java
+RMIC_JAVA = $(BOOTDIR)/bin/java
 # need to treat 64bit solaris differently
 ifeq ($(PLATFORM)-$(LIBARCH), solaris-amd64)
 RMIC_JAVA = $(OUTPUTDIR)/bin/amd64/java
