diff --git a/src/version.c b/src/version.c
index d8ad76b..36350d8 100644
--- a/src/version.c
+++ b/src/version.c
@@ -116,7 +116,10 @@ int read_version_from_subvol_file(char *path_prefix)
		if (fgets(line, LINE_MAX, file) == NULL)
			break;

-		if (strncmp(line,"VERSION_ID=", 11) == 0) {
+		if (strncmp(line,"VERSION_ID=\"", 12) == 0) {
+			v = strtoull(&line[12], NULL, 10);
+			break;
+		} else if (strncmp(line,"VERSION_ID=", 11) == 0) {
			v = strtoull(&line[11], NULL, 10);
			break;
		}
