Disable LDAP support

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Disable LDAP support

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Index: audit-2.3.2/audisp/plugins/Makefile.am
===================================================================
--- audit-2.3.2.orig/audisp/plugins/Makefile.am
+++ audit-2.3.2/audisp/plugins/Makefile.am
@@ -22,8 +22,10 @@
 
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 
-SUBDIRS = builtins zos-remote remote
-#SUBDIRS = builtins zos-remote
+SUBDIRS = builtins remote
+if HAVE_LDAP
+SUBDIRS += zos-remote
+endif
 if HAVE_PRELUDE
 SUBDIRS += prelude
 endif
Index: audit-2.3.2/configure.ac
===================================================================
--- audit-2.3.2.orig/configure.ac
+++ audit-2.3.2/configure.ac
@@ -241,6 +241,12 @@ else
 fi
 AM_CONDITIONAL(HAVE_PRELUDE, test x$have_prelude = xyes)
 
+AC_ARG_WITH(ldap,
+AS_HELP_STRING([--with-ldap],[enable zos-remote plugin, which requires ldap]),
+use_ldap=$withval,
+use_ldap=no)
+AM_CONDITIONAL(HAVE_LDAP, test x$have_ldap = xyes)
+
 AC_MSG_CHECKING(whether to use libwrap)
 AC_ARG_WITH(libwrap,
 [  --with-libwrap[=PATH]   Compile in libwrap (tcp_wrappers) support.],
Index: audit-2.3.2/docs/Makefile.am
===================================================================
--- audit-2.3.2.orig/docs/Makefile.am
+++ audit-2.3.2/docs/Makefile.am
@@ -53,7 +53,9 @@ ausearch_add_expression.3 ausearch_add_t
 ausearch_clear.3 \
 ausearch_next_event.3 ausearch_set_stop.3 \
 autrace.8 get_auditfail_action.3 set_aumessage_mode.3 \
-audispd.8 audispd.conf.5 audispd-zos-remote.8 libaudit.conf.5 \
-augenrules.8 \
-zos-remote.conf.5 
+audispd.8 audispd.conf.5 libaudit.conf.5 \
+augenrules.8
 
+if HAVE_LDAP
+man_MANS += audispd-zos-remote.8 zos-remote.conf.5 
+endif
