From 1f2ad4cdff6b999fcd00193cdd1dc8a35e49c0b2 Mon Sep 17 00:00:00 2001
From: Joshua Lock <joshua.g.lock@intel.com>
Date: Thu, 3 Mar 2016 19:55:41 +0000
Subject: [PATCH] Add configure option to re-enable updating of config files

Signed-off-by: Joshua Lock <joshua.g.lock@intel.com>

Upstream-Status: Pending

---
 configure.ac     | 7 +++++++
 src/heuristics.c | 5 +++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index f94a17d..0063463 100644
--- a/configure.ac
+++ b/configure.ac
@@ -78,6 +78,13 @@ AC_ARG_ENABLE(
 	enable_linux_rootfs_build="no"
 )
 
+AC_ARG_ENABLE(
+        [stateless],
+        AS_HELP_STRING([--disable-stateless],[OS is not stateless, do not ignore configuration files (stateless by default)]),
+        AC_DEFINE(OS_IS_STATELESS,0,[OS is not stateless]),
+        AC_DEFINE(OS_IS_STATELESS,1,[OS is stateless])
+)
+
 AS_IF([test "$enable_linux_btrfs_build" = "yes" -a "$enable_android_build" = "yes"],
       [AC_MSG_ERROR([Cannot enable more than one build variant. Choose a single variant.])])
 
diff --git a/src/heuristics.c b/src/heuristics.c
index 12fb59c..92dbcdc 100644
--- a/src/heuristics.c
+++ b/src/heuristics.c
@@ -27,6 +27,7 @@
 #include <string.h>
 #include <assert.h>
 
+#include <config.h>
 #include <swupd.h>
 
 /* trailing slash is to indicate dir itself is expected to exist, but
@@ -113,8 +114,8 @@ void apply_heuristics(struct file *file)
 
 bool ignore(struct file *file)
 {
-	if ((file->is_config) ||
-	    is_config(file->filename) || // ideally we trust the manifest but short term reapply check here
+	if ((OS_IS_STATELESS && file->is_config) ||
+	    (OS_IS_STATELESS && is_config(file->filename)) || // ideally we trust the manifest but short term reapply check here
 	    (file->is_state) ||
 	    is_state(file->filename) || // ideally we trust the manifest but short term reapply check here
 	    (file->is_boot &&  fix &&  file->is_deleted) || // shouldn't happen
-- 
2.5.0

