From 259d86e64146c3156eccfcce0351a9cdc4714766 Mon Sep 17 00:00:00 2001
From: Jaska Uimonen <jaska.uimonen@intel.com>
Date: Thu, 14 Jan 2016 10:17:43 +0200
Subject: [PATCH] change systemctl path to OE systemctl path

Upstream-Status: Inappropriate

---
 src/scripts.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/scripts.c b/src/scripts.c
index e426272..9bec0f5 100644
--- a/src/scripts.c
+++ b/src/scripts.c
@@ -84,10 +84,10 @@ static void update_triggers(void)
 	int ret;
 	LOG_INFO(NULL, "calling systemd trigger", class_scripts, "");
 
-	ret = system("/usr/bin/systemctl daemon-reload");
+	ret = system("/bin/systemctl daemon-reload");
 	if (ret != 0)
 		LOG_ERROR(NULL, "systemd daemon reload failed", class_scripts, "%d", ret);
-	ret = system("/usr/bin/systemctl restart update-triggers.target");
+	ret = system("/bin/systemctl restart update-triggers.target");
 	if (ret != 0)
 		LOG_ERROR(NULL, "systemd update triggers failed", class_scripts, "%d", ret);
 }
-- 
2.5.0

