From 1dd652640ae4efafd27d4e4ae6d9e77af9b1b7bc Mon Sep 17 00:00:00 2001
From: Dmitry Kasatkin <d.kasatkin@samsung.com>
Date: Thu, 30 Oct 2014 15:26:54 +0200
Subject: [PATCH 02/45] ima: reset EVM status when resetting appraisal status

Signed-off-by: Dmitry Kasatkin <d.kasatkin@samsung.com>
---
 security/integrity/ima/ima_appraise.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/security/integrity/ima/ima_appraise.c b/security/integrity/ima/ima_appraise.c
index 37e1e98..490cc5a 100644
--- a/security/integrity/ima/ima_appraise.c
+++ b/security/integrity/ima/ima_appraise.c
@@ -325,6 +325,7 @@ void ima_inode_post_setattr(struct dentry *dentry)
 	must_appraise = ima_must_appraise(inode, MAY_ACCESS, POST_SETATTR);
 	iint = integrity_iint_find(inode);
 	if (iint) {
+		iint->evm_status = INTEGRITY_UNKNOWN;
 		iint->flags &= ~(IMA_APPRAISE | IMA_APPRAISED |
 				 IMA_APPRAISE_SUBMASK | IMA_APPRAISED_SUBMASK |
 				 IMA_ACTION_FLAGS);
@@ -363,6 +364,7 @@ static void ima_reset_appraise_flags(struct inode *inode, int digsig)
 	if (!iint)
 		return;
 
+	iint->evm_status = INTEGRITY_UNKNOWN;
 	iint->flags &= ~IMA_DONE_MASK;
 	if (digsig)
 		iint->flags |= IMA_DIGSIG;
-- 
2.1.4

