From 14336fd5b1d4efbb71b35cccdd6c2dc906df9132 Mon Sep 17 00:00:00 2001
From: Dmitry Kasatkin <d.kasatkin@samsung.com>
Date: Thu, 30 Oct 2014 15:29:47 +0200
Subject: [PATCH 04/45] ima: reset EVM status along with appraisal status

Signed-off-by: Dmitry Kasatkin <d.kasatkin@samsung.com>
---
 security/integrity/ima/ima_main.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/security/integrity/ima/ima_main.c b/security/integrity/ima/ima_main.c
index 41be9d3..6cd30d0 100644
--- a/security/integrity/ima/ima_main.c
+++ b/security/integrity/ima/ima_main.c
@@ -209,15 +209,19 @@ static int process_measurement(struct file *file, int mask, int function,
 
 	mutex_lock(&iint->mutex);
 
-	if (test_and_clear_bit(IMA_CHANGE_ATTR, &iint->atomic_flags))
+	if (test_and_clear_bit(IMA_CHANGE_ATTR, &iint->atomic_flags)) {
 		/* reset appraisal flags if ima_inode_post_setattr was called */
 		iint->flags &= ~(IMA_APPRAISE | IMA_APPRAISED |
 				 IMA_APPRAISE_SUBMASK | IMA_APPRAISED_SUBMASK |
 				 IMA_ACTION_FLAGS);
+		iint->evm_status = INTEGRITY_UNKNOWN;
+	}
 
-	if (test_and_clear_bit(IMA_CHANGE_XATTR, &iint->atomic_flags))
+	if (test_and_clear_bit(IMA_CHANGE_XATTR, &iint->atomic_flags)) {
 		/* reset all flags if ima_inode_setxattr was called */
 		iint->flags &= ~IMA_DONE_MASK;
+		iint->evm_status = INTEGRITY_UNKNOWN;
+	}
 
 	/* Determine if already appraised/measured based on bitmask
 	 * (IMA_MEASURE, IMA_MEASURED, IMA_XXXX_APPRAISE, IMA_XXXX_APPRAISED,
-- 
2.1.4

