Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@intel.com>

From 580100d7498f861b04e6e727a4fc71bd29df46b6 Mon Sep 17 00:00:00 2001
From: Gwenole Beauchesne <gwenole.beauchesne@intel.com>
Date: Mon, 28 Jul 2014 14:20:33 +0200
Subject: [PATCH] configure: fix build with GStreamer 1.4.0 release.

---
 configure.ac |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index c98d08c..b41d483 100644
--- a/configure.ac
+++ b/configure.ac
@@ -220,7 +220,7 @@ case $GST_API_VERSION in
     GST_PLUGINS_BASE_VERSION_REQUIRED=gst12_plugins_base_version
     GST_PLUGINS_BAD_VERSION_REQUIRED=gst12_plugins_bad_version
     ;;
-1.3)
+1.[[3-4]])
     GST_VERSION_REQUIRED=gst14_version
     GST_PLUGINS_BASE_VERSION_REQUIRED=gst14_plugins_base_version
     GST_PLUGINS_BAD_VERSION_REQUIRED=gst14_plugins_bad_version
@@ -473,7 +473,7 @@ case $GST_API_VERSION in
 0.10)   lt_bias=gst0_vaapi_lt_current_bias;;
 1.0)    lt_bias=gst1_vaapi_lt_current_bias;;
 1.2)    lt_bias=gst2_vaapi_lt_current_bias;;
-1.3)    lt_bias=gst4_vaapi_lt_current_bias;;
+1.[[3-4]]) lt_bias=gst4_vaapi_lt_current_bias;;
 esac
 GST_VAAPI_MAJOR_VERSION=`expr gst_vaapi_lt_current - "$lt_bias"`
 AC_SUBST(GST_VAAPI_MAJOR_VERSION)
-- 
1.7.1
