Fix compilation of the Wayland backend when using GStreamer 0.10.

Upstream-Status: Submitted (https://bugzilla.gnome.org/show_bug.cgi?id=712282)
Signed-off-by: Ross Burton <ross.burton@intel.com>

diff --git a/gst-libs/gst/vaapi/Makefile.am b/gst-libs/gst/vaapi/Makefile.am
index ab605e1..0a3fa38 100644
--- a/gst-libs/gst/vaapi/Makefile.am
+++ b/gst-libs/gst/vaapi/Makefile.am
@@ -364,6 +364,7 @@ libgstvaapi_wayland_@GST_API_VERSION@_la_CFLAGS = \
 	-I$(top_srcdir)/gst-libs		\
 	$(GLIB_CFLAGS)				\
 	$(GST_BASE_CFLAGS)			\
+	$(top_builddir)/gst-libs/gst/video/libgstvaapi-videoutils.la \
 	$(WAYLAND_CFLAGS)			\
 	$(LIBVA_WAYLAND_CFLAGS)			\
 	$(NULL)
