From 841418e5e3f40ee5ae492d606d403feef82cdc1c Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Mon, 22 Feb 2016 22:07:08 -0800
Subject: [PATCH] Fix regression that introduced a directory named '/ '

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>

Upstream-Status: Backport [v3.0+]

---
 src/chroot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/chroot.c b/src/chroot.c
index cb6a04a..58e1f1a 100644
--- a/src/chroot.c
+++ b/src/chroot.c
@@ -41,7 +41,7 @@ void chroot_create_full(int newversion)
 	struct stat buf;
 	char *full_dir;

-	string_or_die(&full_dir, "%s/%i/full/ ", image_dir, newversion);
+	string_or_die(&full_dir, "%s/%i/full/", image_dir, newversion);

 	g_mkdir_with_parents(full_dir, S_IRWXU);

--
2.5.0
