From 6d8d70078c99a733794f4d0311d5a3329ef4dc86 Mon Sep 17 00:00:00 2001
From: Jaska Uimonen <jaska.uimonen@helsinki.fi>
Date: Mon, 27 Jul 2015 10:59:58 +0300
Subject: [PATCH 1/2] fix possible uninitialized array values

Upstream-Status: Accepted

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>
---
 dfa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dfa.c b/dfa.c
index b8b68eb..4249c19 100644
--- a/dfa.c
+++ b/dfa.c
@@ -400,7 +400,7 @@ void ntod ()
 	 * from 1 to CSIZE, so their size must be CSIZE + 1.
 	 */
 	int     duplist[CSIZE + 1], state[CSIZE + 1];
-	int     targfreq[CSIZE + 1], targstate[CSIZE + 1];
+	int     targfreq[CSIZE + 1] = {0}, targstate[CSIZE + 1];
 
 	/* accset needs to be large enough to hold all of the rules present
 	 * in the input, *plus* their YY_TRAILING_HEAD_MASK variants.
-- 
2.1.0

