From b36cef2779ebdd45af1d99ba9e0373548ed5678e Mon Sep 17 00:00:00 2001
From: Ismo Puustinen <ismo.puustinen@intel.com>
Date: Tue, 4 Aug 2015 16:21:40 +0300
Subject: [PATCH 5/6] cgsnapshot.c: free memory after use.

Upstream-status: Submitted

Signed-off-by: Ismo Puustinen <ismo.puustinen@intel.com>

diff --git a/src/tools/cgsnapshot.c b/src/tools/cgsnapshot.c
index a0599c1..806898d 100644
--- a/src/tools/cgsnapshot.c
+++ b/src/tools/cgsnapshot.c
@@ -428,6 +428,8 @@ static int display_cgroup_data(struct cgroup *group,
 				goto err;
 			}
 			fprintf(of, "\t\t%s=\"%s\";\n", output_name, value);
+
+			free(value);
 		}
 		fprintf(of, "\t}\n");
 	}
-- 
2.4.3

