From 2006a5696880e4389016ab68952af3021a44b463 Mon Sep 17 00:00:00 2001
From: Sudarsana Nagineni <sudarsana.nagineni@intel.com>
Date: Fri, 27 Jan 2017 15:50:38 +0200
Subject: [PATCH] Remove iotivity-node dependency from package.json

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>
---
 package.json | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/package.json b/package.json
index 001d0f9..75f9239 100644
--- a/package.json
+++ b/package.json
@@ -9,9 +9,6 @@
   },
   "author": "Sakari Poussa",
   "license": "Apache-2.0",
-  "dependencies": {
-    "iotivity-node": "^1.2.0-1"
-  },
   "devDependencies": {
     "gulp": "^3.8.11",
     "gulp-nodemon": "^2.0.3"
-- 
1.9.1

