From 0b7e8b586169cfabb2e9224209b15c4c913250f4 Mon Sep 17 00:00:00 2001
From: Sudarsana Nagineni <sudarsana.nagineni@intel.com>
Date: Tue, 24 Nov 2015 15:37:25 +0200
Subject: [PATCH] Remove iotivity-node dependency from package.json

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>
---
 package.json | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/package.json b/package.json
index 0679ea9..1c8e1c9 100644
--- a/package.json
+++ b/package.json
@@ -9,9 +9,6 @@
   },
   "author": "Sakari Poussa",
   "license": "Apache-2.0",
-  "dependencies": {
-    "iotivity-node": "^1.0.1-1"
-  },
   "devDependencies": {
     "gulp": "^3.8.11",
     "gulp-nodemon": "^2.0.3"
-- 
1.9.1

