From 08ff50067aac2c70fd561cd565cd87a17b595552 Mon Sep 17 00:00:00 2001
From: Alex Tereschenko <alext.mkrs@gmail.com>
Date: Mon, 6 Jul 2015 18:43:30 +0200
Subject: [PATCH] hcsr04: Fixed compilation error after MRAA enum renaming

Closes #214.

Signed-off-by: Alex Tereschenko <alext.mkrs@gmail.com>
---
 src/hcsr04/hcsr04.cxx | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/hcsr04/hcsr04.cxx b/src/hcsr04/hcsr04.cxx
index aecd8a5..362baac 100644
--- a/src/hcsr04/hcsr04.cxx
+++ b/src/hcsr04/hcsr04.cxx
@@ -49,8 +49,7 @@ HCSR04::HCSR04 (uint8_t triggerPin, uint8_t echoPin, void (*fptr)(void *)) {
     }
 
     mraa_gpio_dir(m_echoPinCtx, MRAA_GPIO_IN);
-    gpio_edge_t edge = MRAA_GPIO_EDGE_BOTH;
-    mraa_gpio_isr (m_echoPinCtx, edge, fptr, NULL);
+    mraa_gpio_isr(m_echoPinCtx, MRAA_GPIO_EDGE_BOTH, fptr, NULL);
 }
 
 HCSR04::~HCSR04 () {
-- 
2.4.4

